src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -1230,7 +1230,9 @@ void DirectoryService::RunConsensusOnDSBlock(bool isRejoin) {
     RejoinAsDS();
   }
 
-  SetState(DSBLOCK_CONSENSUS_PREP);
+  if (m_state != DSBLOCK_CONSENSUS_PREP) {
```

Comment:
Why do we need to check `m_state` here before setting?  Same comment for other change below.

Reply:
It's just about the tidy of the log, if it harms I'll remove the if check

Reply:
Hmm it's OK, we can keep